src/rules/unit-no-unknown/index.js

```diff
@@ -0,0 +1,60 @@
+import { isString } from "lodash"
+import valueParser from "postcss-value-parser"
+
+import {
+  optionsHaveIgnored,
+  isKnownUnit,
+  cssWordIsVariable,
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "unit-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (u) => `Unexpected unknown unit "${u}"`,
+})
+
+export default function (actual, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual }, {
+      actual: options,
+      possible: {
+        ignore: [isString],
```

Comment:
This should be `ignoreUnits`, rather than `ignore`. As per the [original issue](https://github.com/stylelint/stylelint/issues/1004#issuecomment-205350868), which takes into account the [options naming conventions](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#determine-options).